src/rpc_client/mod.rs

```diff
@@ -267,12 +284,38 @@ impl fmt::Display for Url {
     }
 }
 
+#[derive(PartialEq, Eq, Debug)]
+pub enum CommunicationProtocol {
+    Http,
+    Ws,
+}
+
+impl TryFrom<&str> for CommunicationProtocol {
+    type Error = anyhow::Error;
+
+    fn try_from(s: &str) -> Result<Self, Self::Error> {
+        match s {
+            "http" => Ok(Self::Http),
+            "ws" => Ok(Self::Ws),
+            _ => anyhow::bail!("unsupported protocol"),
+        }
+    }
+}
```

Comment:
Could we use https://docs.rs/strum/latest/strum/derive.EnumString.html? We are already doing it at https://github.com/ChainSafe/forest/blob/main/src/utils/misc/mod.rs.

Reply:
Done.